libs/blocks/article-header/article-header.js

```diff
@@ -0,0 +1,198 @@
+  if (img) {
+    img.setAttribute('alt', authorEl.textContent);
+    authorImg.append(img);
+    if (!img.complete) {
+      img.addEventListener('load', () => {
+        authorImg.style.backgroundImage = 'none';
+      });
+      img.addEventListener('error', () => {
+        img.remove();
+      });
+    } else {
+      authorImg.style.backgroundImage = 'none';
+    }
+  }
+}
+
+async function copyToClipboard(button) {
+  try {
+    await navigator.clipboard.writeText(window.location.href);
+    const copiedToClipboard = await replaceKey('copied-to-clipboard', getConfig());
+    button.setAttribute('title', copiedToClipboard);
+    button.setAttribute('alt', copiedToClipboard);
+    button.setAttribute('aria-label', copiedToClipboard);
+
+    const tooltip = createTag('div', { role: 'status', 'aria-live': 'polite', class: 'copied-to-clipboard' }, copiedToClipboard);
+    button.append(tooltip);
+
+    setTimeout(() => {
+      tooltip.remove();
+    }, 3000);
+    button.classList.remove('copy-failure');
+    button.classList.add('copy-success');
+  } catch (e) {
+    button.classList.add('copy-failure');
+    button.classList.remove('copy-success');
+  }
+}
+
+async function buildSharing() {
+  const url = encodeURIComponent(window.location.href);
+  const title = encodeURIComponent(document.querySelector('h1').textContent);
+  const description = encodeURIComponent(getMetadata('description'));
+
+  const platformMap = {
+    twitter: {
+      'data-href': `https://www.twitter.com/share?&url=${url}&text=${title}`,
+      alt: `${await replaceKey('share-twitter', getConfig())}`,
+      'aria-label': `${await replaceKey('share-twitter', getConfig())}`,
+    },
+    linkedin: {
+      'data-type': 'LinkedIn',
+      'data-href': `https://www.linkedin.com/shareArticle?mini=true&url=${url}&title=${title}&summary=${description || ''}`,
+      alt: `${await replaceKey('share-linkedin', getConfig())}`,
+      'aria-label': `${await replaceKey('share-linkedin', getConfig())}`,
+    },
+    facebook: {
+      'data-type': 'Facebook',
+      'data-href': `https://www.facebook.com/sharer/sharer.php?u=${url}`,
+      alt: `${await replaceKey('share-facebook', getConfig())}`,
+      'aria-label': `${await replaceKey('share-facebook', getConfig())}`,
+    },
+    link: {
+      id: 'copy-to-clipboard',
+      alt: `${await replaceKey('copy-to-clipboard', getConfig())}`,
+      'aria-label': `${await replaceKey('copy-to-clipboard', getConfig())}`,
+    },
+  };
+
+  const platforms = Object.keys(platformMap);
+  const svgs = await fetchIcons(getConfig());
+  const anchorTags = platforms.map(async (platform) => {
+    const platformProperties = platformMap[platform];
+    if (platformProperties) {
+      return createTag('a', platformProperties, svgs[platform].cloneNode(true));
+    }
+    return null;
+  });
+
+  const sharing = createTag('div', { class: 'article-byline-sharing' });
+  const allAnchorTags = await Promise.all(anchorTags);
```

Comment:
I don't think you need to use promises as nothing in anchorTags actually awaits anything
```suggestion
  const allAnchorTags = platforms.map((platform) => {
    const platformProperties = platformMap[platform];
    if (platformProperties) {
      return createTag('a', platformProperties, svgs[platform].cloneNode(true));
    }
    return null;
  }).filter(Boolean);

  const sharing = createTag('div', { class: 'article-byline-sharing' });
```